### PR TITLE
Added `pyg_lib.segment_matmul` integration within `RGCNConv`

### PR DESCRIPTION
**Pull request by puririshi98**

### CODE REVIEW

**Comment by Padarn** [comment]
curious why we wouldn't want to make this a dependency of pytorch geometric?

File: torch_geometric/nn/conv/rgcn_conv.py [hunk 1]
```diff
@@ -209,14 +215,23 @@ def forward(self, x: Union[OptTensor, Tuple[OptTensor, Tensor]],
...
+                for i in range(self.num_relations):
+                    tmp = masked_edge_index(edge_index, edge_type == i)
+
+                    if x_l.dtype == torch.long:
+                        out += self.propagate(tmp, x=weight[i, x_l], size=size)
+                    else:
+                        h = self.propagate(tmp, x=x_l, size=size)
+                        out = out + (h @ weight[i])
+            else:
+                edge_type, sort_by_edge_type = torch.sort(edge_type)
```
Comment (rusty1s) [inline torch_geometric/nn/conv/rgcn_conv.py:228]: We should only sorry if necessary, e.g., either via a is_sorted flag or via `edge_type[1:] >= edge_type[:-1].

File: torch_geometric/nn/conv/rgcn_conv.py [hunk 2]
```diff
@@ -209,14 +215,23 @@ def forward(self, x: Union[OptTensor, Tuple[OptTensor, Tensor]],
...
+                edge_index = edge_index[:, sort_by_edge_type]
+                self.edge_ptr = torch.cumsum(
```
Comment (rusty1s) [inline torch_geometric/nn/conv/rgcn_conv.py:230]: Let‘s use ind2ptr for this from torch-sparse.

**Review by rusty1s — APPROVED** [review]
Thanks!

**Comment by rusty1s** [comment]
@Padarn Indeed. Added installation of `pyg_lib` to our CI.

**Pull request merged**